haros/export_manager.py

```diff
@@ -45,6 +45,87 @@ class LoggingObject(object):
 # Export Manager
 ################################################################################
 
+class JUnitExporter(LoggingObject):
+    """
+    Utility class for outputting analysis result data
+    in a JUnit XML format text file.
+    """
+    def export_report(self, datadir, database):
+        """
+        Output the analysis data in a JUnit XML format text file.
+        @param datadir:   [str] The folder / file system path where to store the output.
+        @param database:  [.data.HarosDatabase] Database with analysis result data.
+        """
+        self.log.info("Exporting JUnit XML format report data.")
+        if database.report == None:
+            return
+        report = database.report # .data.AnalysisReport
+        for package_analysis in report.by_package.viewvalues(): # .data.PackageAnalysis
+            out = os.path.join(datadir, package_analysis.package.name + ".xml")
+            f = open(out, "w")
```

Comment:
This should be enclosed in a `try ... finally` or a `with` statement. Is there any reason not to?

Reply:
Thanks, good point. Will add ``try ... finally``.